test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -51,258 +54,518 @@ var (
-	return collectors
 }
 
-func getNodePreparer(nc nodeCase, clientset clientset.Interface) (testutils.TestNodePreparer, error) {
+func getNodePreparer(prefix string, cno *createNodesOp, clientset clientset.Interface) (testutils.TestNodePreparer, error) {
 	var nodeStrategy testutils.PrepareNodeStrategy = &testutils.TrivialNodePrepareStrategy{}
-	if nc.NodeAllocatableStrategy != nil {
-		nodeStrategy = nc.NodeAllocatableStrategy
-	} else if nc.LabelNodePrepareStrategy != nil {
-		nodeStrategy = nc.LabelNodePrepareStrategy
-	} else if nc.UniqueNodeLabelStrategy != nil {
-		nodeStrategy = nc.UniqueNodeLabelStrategy
+	if cno.NodeAllocatableStrategy != nil {
+		nodeStrategy = cno.NodeAllocatableStrategy
+	} else if cno.LabelNodePrepareStrategy != nil {
+		nodeStrategy = cno.LabelNodePrepareStrategy
+	} else if cno.UniqueNodeLabelStrategy != nil {
+		nodeStrategy = cno.UniqueNodeLabelStrategy
 	}
 
-	if nc.NodeTemplatePath != nil {
-		node, err := getNodeSpecFromFile(nc.NodeTemplatePath)
+	if cno.NodeTemplatePath != nil {
+		node, err := getNodeSpecFromFile(cno.NodeTemplatePath)
 		if err != nil {
 			return nil, err
 		}
 		return framework.NewIntegrationTestNodePreparerWithNodeSpec(
 			clientset,
-			[]testutils.CountToStrategy{{Count: nc.Num, Strategy: nodeStrategy}},
+			[]testutils.CountToStrategy{{Count: cno.CreateNodes.(int), Strategy: nodeStrategy}},
 			node,
 		), nil
 	}
 	return framework.NewIntegrationTestNodePreparer(
 		clientset,
-		[]testutils.CountToStrategy{{Count: nc.Num, Strategy: nodeStrategy}},
-		"scheduler-perf-",
+		[]testutils.CountToStrategy{{Count: cno.CreateNodes.(int), Strategy: nodeStrategy}},
+		prefix,
 	), nil
 }
 
-func createPods(ns string, pc podCase, clientset clientset.Interface) error {
-	strategy, err := getPodStrategy(pc)
+func createPods(namespace string, cpo *createPodsOp, clientset clientset.Interface) error {
+	strategy, err := getPodStrategy(cpo)
 	if err != nil {
 		return err
 	}
 	config := testutils.NewTestPodCreatorConfig()
-	config.AddStrategy(ns, pc.Num, strategy)
+	config.AddStrategy(namespace, cpo.CreatePods.(int), strategy)
 	podCreator := testutils.NewTestPodCreator(clientset, config)
 	return podCreator.CreatePods()
 }
 
-func getPodStrategy(pc podCase) (testutils.TestPodCreateStrategy, error) {
-	basePod := makeBasePod()
-	if pc.PodTemplatePath != nil {
-		var err error
-		basePod, err = getPodSpecFromFile(pc.PodTemplatePath)
+// barrierOne blocks until all pods in the given namespace are scheduled.
+func barrierOne(podInformer coreinformers.PodInformer, name string, namespace string, wantCount int) error {
+	for {
+		scheduled, err := getScheduledPods(podInformer, namespace)
 		if err != nil {
-			return nil, err
+			return err
 		}
+		if len(scheduled) >= wantCount {
+			break
+		}
+		klog.Infof("%s: namespace %s: got %d existing pods, want %d", name, namespace, len(scheduled), wantCount)
+		time.Sleep(1 * time.Second)
```

Comment:
we need a timeout. You can reuse functions in `k8s.io/apimachinery/pkg/util/wait`